File: ios/RoutineChart/RoutineChart/Core/Data/Local/Database/SyncCursor.swift
//
//  SyncCursor.swift
//  RoutineChart
//
//  Created for Phase 3.1: Sync Infrastructure
//  Tracks last sync timestamp per collection for pull cursor pattern
//

import Foundation
import GRDB

/// Tracks the last sync timestamp for a specific collection
struct SyncCursor: Identifiable, Equatable, Sendable {
    let id: String // collection name (e.g., "routines", "events")
    let collection: String
    let lastSyncedAt: Date
    
    init(collection: String, lastSyncedAt: Date) {
        self.id = collection
        self.collection = collection
        self.lastSyncedAt = lastSyncedAt
    }
}

File: android/app/src/main/java/com/HammersTech/RoutineChart/core/domain/models/SyncCursor.kt
package com.HammersTech.RoutineChart.core.domain.models

import java.time.Instant

/**
 * Domain model for sync cursor
 * Tracks the last sync timestamp for a specific collection
 * Phase 3.1: Sync Infrastructure
 */
data class SyncCursor(
    val collection: String, // e.g., "routines", "events", "steps", "assignments", "children"
    val lastSyncedAt: Instant,
)
